test/test_displacement.py

```diff
@@ -0,0 +1,96 @@
+#  Copyright (c) 2025 National Land Survey of Finland (Maanmittauslaitos)
+#
+#  This file is part of geogen-algorithms.
+#
+#  This source code is licensed under the MIT license found in the
+#  LICENSE file in the root directory of this source tree.
+
+import geopandas as gpd
+import pytest
+from pandas.testing import assert_frame_equal
+from shapely import equals_exact
+from shapely.geometry import Point
+
+from geogenalg import displacement
+
+
+@pytest.mark.parametrize(
+    ("input_gdf", "threshold", "expected_gdf"),
+    [
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(0, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(2, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(2, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(0.5, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(-0.25, 0), Point(0.75, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(1, 0), Point(2, 0)]),
+            2.0,
+            gpd.GeoDataFrame(geometry=[Point(-0.5, 0), Point(1, 0), Point(2.5, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(0.5, 0), Point(5, 5)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(-0.25, 0), Point(0.75, 0), Point(5, 5)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(0, 0), Point(0, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(0, 0), Point(0, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(1.0, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(1.0, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(
+                {"id": [5, 4, 6], "category": ["Z", "X", "Y"]},
+                geometry=[Point(0, 0), Point(0.5, 0), Point(5, 5)],
+            ),
+            1.0,
+            gpd.GeoDataFrame(
+                {"id": [5, 4, 6], "category": ["Z", "X", "Y"]},
+                geometry=[Point(-0.25, 0), Point(0.75, 0), Point(5, 5)],
+            ),
+        ),
+    ],
+    ids=[
+        "single_point",
```

Comment:
The test cases are good. You could make them even easier to understand by always descriping clearly what should happen here in the id. For example:

- single point should remain unmodified
- two far apart should remain unmodified
- two close should be displaced
- and so on

Oh and these id's don't have be snake_case. Of course you can use snake_case if you want to but it's not a must. Sometimes hyphens are used (for-example-like-this) but to my understanding it's also totally fine to just write them normally without any punctuation marks if you like.